### PR TITLE
Player:onTradeCompleted(...)

### PR DESCRIPTION
**Pull request by Zbizu**
new player event triggering after trade was successful

(onTradeAccept triggers whether the trade is successful or not so they're different)

### CODE REVIEW

**Comment by Zbizu** [comment]
personal thoughts:
most of `Events::eventPlayerOnTradeSuccess` function content is same as tradeRequest and tradeAccept

There is probably a way to optimize it, but I'm not aware of such.

**Comment by nekiro** [comment]
shouldnt this event be more unified like OnTradeCompleted and push if success or not in parameter?

**Comment by mackerel225** [comment]
> 
> 
> shouldnt this event be more unified like OnTradeCompleted and push success variable as parameter?

My initial thought, with this you could remove onTradeAccept, or keep them both.

**Comment by Zbizu** [comment]
great idea

I will edit it in a few hours.

File: src/events.cpp [hunk 1]
```diff
@@ -820,6 +822,41 @@ bool Events::eventPlayerOnTradeAccept(Player* player, Player* target, Item* item
 	return scriptInterface.callFunction(4);
 }
 
+void Events::eventPlayerOnTradeCompleted(Player* player, Player* target, Item* item, Item* targetItem, bool isSuccess)
+{
+	// Player:onTradeSuccess(target, item, targetItem)
```
Comment (nekiro) [inline src/events.cpp:827]: you are missing rename and parameter in that comment

**Comment by Zbizu** [comment]
There we go

**Comment by nekiro** [comment]
@Zbizu onTradeAccept is called before all that happens so you can prevent from trading some stuff etc, so no thats different event, used for different purposes.

**Comment by Zbizu** [comment]
possible conflicts: #2966 #2969

**Pull request merged**